src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs

```diff
@@ -133,6 +133,54 @@ public void MoveCaret(int position)
                 view.Caret.MoveTo(point);
             });
 
+        public void EncapsulateField()
+            => GetDTE().ExecuteCommand(WellKnownCommandNames.Refactor_EncapsulateField);
+
+        public void WaitForPreviewChanges()
+            => Retry(FindPreviewChangesDialog, v => v != null, TimeSpan.FromMilliseconds(100));
+
+        public bool IsPreviewChangesShowing()
+            => FindPreviewChangesDialog() != null;
+
+        private static AutomationElement FindPreviewChangesDialog()
+        {
+            var vsAutomationElement = AutomationElement.FromHandle(new IntPtr(GetDTE().MainWindow.HWnd));
+            Condition condition = new PropertyCondition(AutomationElement.ControlTypeProperty, ControlType.Window);
+            var children = vsAutomationElement.FindAll(TreeScope.Children, condition);
+            foreach (AutomationElement child in children)
+            {
+                if (child.Current.Name.Contains("Preview Changes"))
+                {
+                    return child;
+                }
+            }
+
+            return null;
+        }
+
+        public void ClickPreviewChangesCancel()
+        {
+            ClickPreviewChangesButton("CancelButton");
+        }
+
+        public void ClickPreviewChangesOK()
+        {
+            ClickPreviewChangesButton("OkButton");
+        }
+
+        private static void ClickPreviewChangesButton(string buttonAutomationId)
```

Comment:
I don't think you should need any of this. There are already helpers for clicking buttons in dialogs. Why is this one special?